### PR TITLE
truncate first to release disk space to OS

### PR DESCRIPTION
**Pull request by sravfeyn**
@mkangia

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/tasks.py [hunk 1]
```diff
@@ -102,8 +102,10 @@ def prune_synclogs():
             year=year,
             week="%02d" % week
         )
+        truncate_query = "TRUNCATE TABLE {}".format(table_name)
         drop_query = "DROP TABLE IF EXISTS {}".format(table_name)
         db = router.db_for_write(SyncLogSQL)
         with connections[db].cursor() as cursor:
+            cursor.execute(truncate_query)
```
Comment (mkangia) [inline corehq/ex-submodules/casexml/apps/phone/tasks.py]: if we don't truncate the space is not released on drop?

Comment (mkangia) [reply corehq/ex-submodules/casexml/apps/phone/tasks.py]: Does it need "IF exists" check too?

Comment (sravfeyn) [reply corehq/ex-submodules/casexml/apps/phone/tasks.py]: > if we don't truncate the space is not released on drop?
Right, you may read the documentation https://www.postgresql.org/docs/9.1/sql-truncate.html. 

Also it doesn't support the `IF exists` check, given that there is a row in the table, the check should not be necessary anyway.

**Review by mkangia — APPROVED** [review]

**Review by snopoke — COMMENTED** [review]
`DROP TABLE` should also release the resource immediately since it actually deletes the table files on disc. `TRUNCATE` is only necessary if you want to keep the table.

**Comment by sravfeyn** [comment]
> DROP TABLE should also release the resource immediately since it actually deletes the table files on disc. TRUNCATE is only necessary if you want to keep the table.

If you are sure, then I will close this. There were some notes where where I saw TRUNCATE is preferred. Perhaps it's faster on larger tables.

**Comment by snopoke** [comment]
I'm 100% sure that DROP is better than TRUNCATE unless you want to keep the table definition which we do not want to do here. It is particularly unnecessary if you are immediately going to drop the table anyway.

**Comment by sravfeyn** [comment]
Great, thanks. I am closing this then

**Pull request closed without merging**